applications/FemToDemApplication/custom_processes/expand_wet_nodes_process.cpp

```diff
@@ -92,4 +97,58 @@ void ExpandWetNodesProcess::ExpandWetNodes(
 /***********************************************************************************/
 /***********************************************************************************/
 
-}
\ No newline at end of file
+void ExpandWetNodesProcess::ExpandWetNodesIfTheyAreSkin()
+{
+    Parameters skin_process_parameters = Parameters(R"(
+    {
+        "name_auxiliar_model_part"              : "SkinModelPart",
+        "name_auxiliar_condition"               : "Condition",
+        "list_model_parts_to_assign_conditions" : [],
+        "echo_level"                            : 0
+    })");
+
+    auto skin_process = SkinDetectionProcess<2>(mrModelPart, skin_process_parameters); // TODO what's up in 3d?
+    skin_process.Execute();
+	auto& r_sub_model_part = mrModelPart.GetSubModelPart("SkinModelPart");
+
+    // Let's assign the flag IS_SKIN to the nodes
+    for (auto it_node = r_sub_model_part.Nodes().ptr_begin(); it_node != r_sub_model_part.Nodes().ptr_end(); ++it_node) {
+        (*it_node)->SetValue(IS_SKIN, true);
+    }
+
+    int expanded_elements = 1;
+    while (expanded_elements > 0) {
+        expanded_elements = 0;
+        for (auto it_elem = mrModelPart.Elements().ptr_begin(); it_elem != mrModelPart.Elements().ptr_end(); ++it_elem) {
+            auto& r_geometry = (*it_elem)->GetGeometry();
+
+            bool element_done = (*it_elem)->GetValue(PRESSURE_EXPANDED);
+            int number_of_wet_nodes, node_pressure_id;
+            // Indicator to reconstruct the Pressure afterwards
+            auto& r_process_info = mrModelPart.GetProcessInfo();
+
+            if (this->ElementHasWetNodes(it_elem, node_pressure_id, number_of_wet_nodes) && element_done == false) {
```

Comment:
```suggestion
            if (this->ElementHasWetNodes(it_elem, node_pressure_id, number_of_wet_nodes) && !element_done) {
```

Reply:
so elegant! thank you

Reply:
![](https://media1.tenor.com/images/cf1bfcc77625c6d5e7436ada3a92ec00/tenor.gif?itemid=5024340)

Reply:
BTW you can commit suggestions directly from GitHub

Reply:
wow I didn't know that :)